### PR TITLE
feat: ORV2-1749 Tighten up user navigation in frontend

### PR DESCRIPTION
**Pull request by krishnan-aot**
# Description

Please provide a summary of the change and the issue fixed. Please include relevant context. List dependency changes.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://onroutebc-1006-frontend.apps.silver.devops.gov.bc.ca)
- [Vehicles](https://onroutebc-1006-vehicles.apps.silver.devops.gov.bc.ca/api)
- [Dops](https://onroutebc-1006-dops.apps.silver.devops.gov.bc.ca/api)
- [TPS-Migration](https://onroutebc-1006-tps-migration.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/onroutebc/actions/workflows/analysis.yml)

After merge, new images are promoted to:
- [Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge.yml)

### CODE REVIEW

File: frontend/src/common/authentication/util.ts [hunk 1]
```diff
@@ -44,3 +50,59 @@ export const DoesUserHaveRoleWithContext = (requiredRole: Optional<string>) => {
...
+  userRoles,
+  allowedRoles,
+}: {
+  userRoles: Optional<UserRolesType[]>;
+  allowedRoles: UserRolesType[];
+}) => {
+  return (
+    userRoles &&
+    userRoles.length &&
+    userRoles.some((userRole) => allowedRoles.includes(userRole))
```
Comment (erikataot) [inline frontend/src/common/authentication/util.ts]: might be able to replace all 3 lines with:
```
userRoles?.some((userRole) => allowedRoles?.includes(userRole))
```

File: frontend/src/features/permits/pages/Amend/AmendPermit.tsx [hunk 2]
```diff
@@ -9,7 +9,7 @@ import { useMultiStepForm } from "../../hooks/useMultiStepForm";
 import { AmendPermitContext } from "./context/AmendPermitContext";
 import { Loading } from "../../../../common/pages/Loading";
 import OnRouteBCContext from "../../../../common/authentication/OnRouteBCContext";
-import { USER_AUTH_GROUP } from "../../../manageProfile/types/userManagement.d";
+import { USER_AUTH_GROUP } from "../../../../common/authentication/types";
```
Comment (krishnan-aot) [inline frontend/src/features/permits/pages/Amend/AmendPermit.tsx:12]: I'm consolidating the User auth group to the common authentication folder. However, there are more changes required. So, I figured it deserves its own PR rather than cram this PR with more changes.

File: frontend/src/common/authentication/auth-walls/BCeIDAuthWall.tsx
Comment (krishnan-aot) [inline frontend/src/common/authentication/auth-walls/BCeIDAuthWall.tsx:1]: BCeIDAuthWall ensures that only properly set up BCeID users are able to get through.
New BCeID users will be taken to welcome page.
IDIR users are taken to Unexpected Error page - This is until we have IDIR user acting as a company developed.

File: frontend/src/common/authentication/auth-walls/IDIRAuthWall.tsx
Comment (krishnan-aot) [inline frontend/src/common/authentication/auth-walls/IDIRAuthWall.tsx:1]: IDIRAuthWall ensures that only IDIR users with appropriate userAuthGroup (PPC_CLERK, SYSADMIN, etc.) have access to a page.

Anyone else is taken to unauthorized page.

File: frontend/src/common/authentication/auth-walls/NewBCeIDAuthWall.tsx
Comment (krishnan-aot) [inline frontend/src/common/authentication/auth-walls/NewBCeIDAuthWall.tsx:1]: Similar in nature to BCeIDAuthWall but applies only for new BCeID users. It ensures that only those users who are completely new remain on the appropriate page.

IDIR users are directed to unexpected error page till we enable IDIR user create company (if applicable, i.e.).

Existing BCeID users will be redirected to the applications page.

File: frontend/src/common/authentication/auth-walls/NewBCeIDAuthWall.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+  const { isNewBCeIDUser, companyId, userDetails } =
+    useContext(OnRouteBCContext);
+
+  const userIDP = userFromToken?.profile?.identity_provider as string;
+  const location = useLocation();
+
+  useEffect(() => {
+    if (!isAuthLoading && isAuthenticated) {
+      console.log("user context state update");
```
Comment (erikataot) [inline frontend/src/common/authentication/auth-walls/NewBCeIDAuthWall.tsx]:
```suggestion

```

File: frontend/src/features/permits/pages/Amend/AmendPermit.tsx (hunk 2 above)
Comment (erikataot) [reply frontend/src/features/permits/pages/Amend/AmendPermit.tsx:12]: that would definitely be a good change 👍

**Review by erikataot — APPROVED** [review]
as far as I've been able to test, it's holding up - nice
